xbmc/interfaces/python/PythonInvoker.cpp

```diff
@@ -154,17 +154,11 @@ bool CPythonInvoker::execute(const std::string& script, std::vector<std::wstring
   {
     if (!m_threadState)
     {
-#if PY_VERSION_HEX < 0x03070000
-      // this is a TOTAL hack. We need the GIL but we need to borrow a PyThreadState in order to get it
-      // as of Python 3.2 since PyEval_AcquireLock is deprecated
-      extern PyThreadState* savestate;
-      PyEval_RestoreThread(savestate);
-#else
-      PyThreadState* ts = PyInterpreterState_ThreadHead(PyInterpreterState_Main());
-      PyEval_RestoreThread(ts);
-#endif
+  // Safely acquire the GIL without borrowing another thread's state
+  PyEval_AcquireLock();
```

Comment:
Is this a good idea? `PyEval_AcquireLock` has been deprecated in 2011 and  `PyEval_SaveThread`/`PyEval_RestoreThread` is the recommended current API (https://docs.python.org/dev/whatsnew/3.2.html#porting-to-python-3-2). I'm skeptical that this is the way forward.